lib/honeycomb/propagation/aws.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require "base64"
+require "json"
+require "uri"
+
+module Honeycomb
+  # Parsing and propagation for AWS trace headers
+  module AWSPropagation
+    # Parse trace headers
+    module UnmarshalTraceContext
+      def parse(serialized_trace)
+        unless serialized_trace.nil?
+          split = serialized_trace.split(";")
+
+          trace_id, parent_span_id, trace_fields = get_fields(split)
+
+          parent_span_id = trace_id if parent_span_id.nil?
+
+          trace_fields = {} if trace_fields.nil?
+
+          if !trace_id.nil? && !parent_span_id.nil?
+            return [trace_id, parent_span_id, trace_fields]
+          end
+        end
+
+        [nil, nil, nil, nil]
+      end
+
+      def get_fields(fields)
+        trace_fields = {}
+        fields.each do |entry|
+          key, value = entry.split("=", 2)
+          case key.downcase
+          when "root"
+            trace_id = value
+          when "parent"
+            parent_span_id = value
```

Comment:
We need prefer self over parent here, and only use parent if parent ID hasn't already been set.

```ruby
            parent_span_id = value if parent_span_id.nil?
```

Reply:
My intention here (though far from perfect, might make more sense as an `if`) was that putting `self` _below_ parent in a case statement will mean that `self` will overwrite anything set by `parent`. So if there's a `parent` and no `self`, `parent` will get assigned, same deal vice versa, and with both, the final assignment wins.